### PR TITLE
Set segments to array type

### PR DESCRIPTION
**Pull request by Akinari1087**
# Description

Changes segments to type Array.

## Issues Resolved

Resolves: https://github.com/sous-chefs/consul/issues/549

## Check List

- [ Enterprise License ] All tests pass. See TESTING.md for details.
- [ N/A ] New functionality includes testing.
- [ N/A ] New functionality has been documented in the README if applicable.

Tested and working with our enterprise license.  Would love to be able to reference this repo instead of our fork.

### CODE REVIEW

**Comment by kitchen-porter** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">Please include a CHANGELOG entry.</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This Pull Request is probably missing tests.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: metadata.rb [hunk 1]
```diff
@@ -6,7 +6,7 @@
 source_url        'https://github.com/sous-chefs/consul'
 issues_url        'https://github.com/sous-chefs/consul/issues'
 chef_version      '>= 13.4'
-version           '4.0.3'
+version           '4.0.4'
```
Comment (ramereth) [inline metadata.rb:9]: Please don't bump the version in the PR. We'll do that separately.

Comment (Akinari1087) [reply metadata.rb]: Thanks, done.

File: CHANGELOG.md [hunk 2]
```diff
@@ -8,6 +8,7 @@
 - Fix Dokken configuration for CentOS 8 and Ubuntu 16.04
 - Enable enforce_idempotency checks
 - Remove support for CentOS 6, Ubuntu < 16.04
+- Change segments to Array type #549
```
Comment (ramereth) [inline CHANGELOG.md:11]: Since I just did a release this morning, can you please move this up into a new ``## Unreleased`` section?

**Comment by kitchen-porter** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This Pull Request is probably missing tests.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: CHANGELOG.md (hunk 2 above)
Comment (Akinari1087) [reply CHANGELOG.md]: I missed that was changed in the merge.  Fixed.

**Comment by ramereth** [comment]
@Akinari1087 I'd like to test this out locally first. Can you please give me an example of a recipe I could use to test this?

Thanks-

**Comment by Akinari1087** [comment]
> @Akinari1087 I'd like to test this out locally first. Can you please give me an example of a recipe I could use to test this?
> 
> Thanks-

You'll only be able to validate startup with an enterprise license if you have one.

Using kitchen, it's difference of:
As [Hash,Mash]:
```
    attributes:
      consul:
        config:
          server: true
          segments:
            name: "test-local"
            port: 8808
```

As Array (Fix):
```
    attributes:
      consul:
        config:
          server: true
          segments:
          - name: "test-local"
            port: 8808
```

Existing result:
```
Aug 31 21:07:54 os-server-1-ubuntu-1804 systemd[1]: Started consul.
Aug 31 21:07:54 os-server-1-ubuntu-1804 consul[15032]: ==> Error parsing /etc/consul/consul.json: 1 error(s) decoding:
Aug 31 21:07:54 os-server-1-ubuntu-1804 consul[15032]: * 'segments': source data must be an array or slice, got map
Aug 31 21:07:54 os-server-1-ubuntu-1804 systemd[1]: consul.service: Main process exited, code=exited, status=1/FAILURE
Aug 31 21:07:54 os-server-1-ubuntu-1804 systemd[1]: consul.service: Failed with result 'exit-code'.
Aug 31 21:07:54 os-server-1-ubuntu-1804 systemd[1]: consul.service: Service hold-off time over, scheduling restart.
Aug 31 21:07:54 os-server-1-ubuntu-1804 systemd[1]: consul.service: Scheduled restart job, restart counter is at 3.
Aug 31 21:07:54 os-server-1-ubuntu-1804 systemd[1]: Stopped consul.
```

Fix result (No license):
```
Aug 31 20:06:15 os-server-1-ubuntu-1804 consul[15083]: ==> Network segments are not supported in this version of Consul
Aug 31 20:06:15 os-server-1-ubuntu-1804 systemd[1]: consul.service: Main process exited, code=exited, status=1/FAILURE
Aug 31 20:06:15 os-server-1-ubuntu-1804 systemd[1]: consul.service: Failed with result 'exit-code'.
Aug 31 20:06:15 os-server-1-ubuntu-1804 systemd[1]: consul.service: Service hold-off time over, scheduling restart.
Aug 31 20:06:15 os-server-1-ubuntu-1804 systemd[1]: consul.service: Scheduled restart job, restart counter is at 5.
Aug 31 20:06:15 os-server-1-ubuntu-1804 systemd[1]: Stopped consul.
```

**Comment by ramereth** [comment]
released as 4.0.5

**Pull request merged**